indico/modules/events/abstracts/models/abstracts.py

```diff
@@ -82,54 +169,136 @@ def __table_args__(cls):
             lazy=True
```

Comment:
this one should also have a custom `primaryjoin` condition to exclude deleted ones